pkg/network/java/testutil/JavaClientSimulator/src/main/java/Clients/JavaClients.java

```diff
@@ -19,118 +21,149 @@
+        try {
+            return ClientType.valueOf(clientTypeArg.toLowerCase());
+        } catch (IllegalArgumentException e) {
+            return ClientType.unsupported;
         }
     }
 
-    public static void HTTPApacheClientExample(int iterations, String url) throws InterruptedException, IOException {
-        if (iterations == -1){
-            while (true){
-                CloseableHttpClient httpClient = HttpClients.createDefault();
-                try {
-                    HttpGet request = new HttpGet("https://"+url);
-                    CloseableHttpResponse response = httpClient.execute(request);
-                    System.out.println("response: " + response);
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-                finally {
-                    httpClient.close();
-                    Thread.sleep(1000);
-                }
+    public static void executeCallback(String clientTypeArg, int iterations, long sleepInterval, String url) throws InterruptedException, ParseException {
 
-            }
-        } else{
-            while (iterations > 0) {
-                CloseableHttpClient httpClient = HttpClients.createDefault();
-                try {
-                    HttpGet request = new HttpGet("https://"+url);
-                    CloseableHttpResponse response = httpClient.execute(request);
-                    System.out.println("response: " + response);
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-                finally {
-                    httpClient.close();
-                    iterations--;
-                }
-            }
+        ClientType clientType = getClientType(clientTypeArg);
+
+        System.out.println("URL: " + url);
+        System.out.println("Iterations: " + iterations);
+        System.out.println("Interval: " + sleepInterval);
+
+        Runnable callback = null;
+
+        // Execute handler based on client type
+        switch (clientType) {
+            case apache:
+                System.out.println("Executing handler for Apache Http client:");
+                callback = () -> {
+                    try {
+                        JavaClients.HTTPApacheClientExample(url);
+                    } catch (IOException e) {
+
+                    }
+                };
+                executeCallbackLogic(iterations, sleepInterval, url, callback);
+                break;
+            case okhttp:
+                System.out.println("Executing handler for OkHttp client:");
+                callback = () -> {
+                    try {
+                        JavaClients.OkHttpClient(url);
+                    } catch (IOException e) {
+
+                    }
+                };
+                executeCallbackLogic(iterations, sleepInterval, url, callback);
+                break;
```

Comment:
suggestion:
just set here the callback, break, and outside the switch case call ` executeCallbackLogic(iterations, sleepInterval, url, callback);` as it is being duplicated for all cases